.mentat/precommit.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Run prettier to format code
+npm run format
+
+# Run ESLint with auto-fix
+npm run lint -- --fix
+
+# Build the project (client and server)
```

Comment:
lets actually remnove the things below this line. just have format and lint/fix.